libs/impeller2/src/types.rs

```diff
@@ -825,7 +825,7 @@ impl<B: IoBuf> MsgBuf<B> {
     }
 
     pub fn try_parse<'a, T: Msg + Deserialize<'a> + 'a>(&'a self) -> Option<Result<T, Error>> {
-        if T::ID == self.id {
```

Comment:
Looks like this was completely broken before. Is `try_parse` just never being used?